examples/ai-ml/ray/main.tf

```diff
@@ -0,0 +1,365 @@
+# EKS Blueprints AddOns
+#---------------------------------------------------------------
+module "eks_blueprints_kubernetes_addons" {
+  source = "../../../modules/kubernetes-addons"
+
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider    = module.eks_blueprints.oidc_provider
+  eks_cluster_version  = module.eks_blueprints.eks_cluster_version
+  eks_cluster_domain   = var.eks_cluster_domain
+
+  # Add-Ons
+  enable_kuberay_operator             = true
+  enable_ingress_nginx                = true
+  enable_aws_load_balancer_controller = true
+  enable_external_dns                 = var.eks_cluster_domain == null ? false : true
+  enable_kube_prometheus_stack        = true
+
+  # Add-on customizations
+  ingress_nginx_helm_config = {
+    values = [templatefile("${path.module}/helm-values/nginx-values.yaml", {
+      hostname     = var.eks_cluster_domain
+      ssl_cert_arn = var.acm_certificate_domain == null ? null : data.aws_acm_certificate.issued[0].arn
+    })]
+  }
+  kube_prometheus_stack_helm_config = {
+    values = [templatefile("${path.module}/helm-values/kube-stack-prometheus-values.yaml", {
+      hostname = var.eks_cluster_domain
+    })]
+    set_sensitive = [
+      {
+        name  = "grafana.adminPassword"
+        value = data.aws_secretsmanager_secret_version.admin_password_version.secret_string
+      }
+    ]
+  }
+
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Deploy Ray Cluster Resources
+#---------------------------------------------------------------
+resource "aws_kms_key" "objects" {
+  enable_key_rotation     = true
+  description             = "KMS key is used to encrypt bucket objects"
+  deletion_window_in_days = 7
+}
+
+
+#tfsec:ignore:aws-s3-enable-bucket-logging tfsec:ignore:aws-s3-enable-versioning
+module "s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "v3.3.0"
+
+  bucket_prefix           = "ray-demo-models-"
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.objects.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+}
+
+data "aws_iam_policy_document" "irsa_policy" {
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}"]
+  }
+  statement {
+    actions   = ["s3:*Object"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}/*"]
+  }
+  statement {
+    actions = [
+      "ssm:PutParameter",
+      "ssm:DeleteParameter",
+      "ssm:GetParameterHistory",
+      "ssm:GetParametersByPath",
+      "ssm:GetParameters",
+      "ssm:GetParameter",
+      "ssm:DeleteParameters",
+      "ssm:DescribeParameters"
+    ]
+    resources = ["*"]
```

Comment:
what SSM parameters are in play here?

Reply:
oh now I see down below its used for checkpointing. maybe a note about scoping it to a prefix but that it will need to match whats used in the codebase (add a local variable and pass to both the policy here and the code via an env var?)